log/README.md

```diff
@@ -0,0 +1,159 @@
+# Contextual Logging
+
+## Log
+This library is a contextual logging library that makes use of context as part of the logging process. It is designed to make development easier by using the context variable to log instead of using a single global logger or passing a logger to every function.
+
+## Why use this library?
+### Do a lot of things in one operation
+The library focuses on doing multiple operations, whether it is adding more fields, logger, or logger configuration, in one chained operation. This makes setup and logging very simple, especially when they involve `Fields`.
+
+### Shallow Context Tree
+`Fields` are stored in the context tree when you use the `Onto` method. Doing many things in one operation allows you to produce a shallow context tree as you do not need to add `Fields` one-by-one. By finalizing the `Fields` operation using `Onto`, it ensures that it will only add all the provided `Fields` once. This is very beneficial when you have several fields to be logged and the bigger the system you have, it is highly likely that the number of fields will grow.
```

Comment:
I think the phrasing of the last sentence is is a bit awkward. Would probably go with something like
> This is extremely beneficial when several fields must be logged, which is common in large and complex codebases.

instead of

> This is very beneficial when you have several fields to be logged and the bigger the system you have, it is highly likely that the number of fields will grow.
If that's the message you were trying to get across